samcli/config/config.py

```diff
@@ -0,0 +1,233 @@
+        }
+        > config = Config().load(schema=schema)
+
+    Attributes
+    ----------
+    __user_config_file : Path
+        User configuration file path (.samrc).
+    __project_config_file : Path
+        Project configuration file path (current dir/.samrc).
+    config_file : dict
+        Path to configuration file chosen.
+    config: dict
+        SAMRC configuration
+    schema: dict
+        JSONSchema for Configuration validation
+    default_schema_file: Path
+        Path to default SAMRC JSONSchema file
+    default_schema: Dict
+        Default JSONSchema for SAMRC validation
+    """
+
+    def __init__(self, schema=None):
+        self.__user_config_file = None
+        self.__project_config_file = None
+        self.config_file = None
+        self.config = None
+        self.default_schema_file = Path(__file__).parent.joinpath('schema.json')
+        self.default_schema = json.load(Path.open(self.default_schema_file))
+        self.schema = self.default_schema if not schema else schema
+
+    def load(self):
+        """Load configuration file and expose as a dictionary
+
+        Returns
+        -------
+        Dict
+            SAMRC configuration
+        """
+
+        LOG.debug("Looking for SAMRC before attempting to load")
+        user_config, project_config = self.__find_config()
+
+        if user_config is not None and project_config is not None:
+            LOG.debug("Found more than one SAMRC; Merging...")
+            LOG.debug("%s", user_config)
+            LOG.debug("%s", project_config)
+            user_config = self.__read_config(user_config)
+            project_config = self.__read_config(project_config)
+            self.config = self.merge_config(user_config, project_config)
+        elif user_config is not None and project_config is None:
+            LOG.debug("Found one SAMRC -- User specific")
+            LOG.debug("%s", user_config)
+            self.config = self.__read_config(user_config)
+        elif project_config is not None and user_config is None:
+            LOG.debug("Found one SAMRC -- Project specific")
+            LOG.debug("%s", project_config)
+            self.config = self.__read_config(project_config)
+
+        LOG.debug("SAMRC is ready to be validated")
+        LOG.debug("%s", self.config)
+        self.validate_config(self.config, self.schema)
+
+        return self.config
+
+    def validate_config(self, config, schema):
+        """Validate configuration against schema
+
+        Parameters
+        ----------
+        config : dict
+            SAMRC configuration
+        schema : dict
+            JSONSchema to validate against
+
+        Returns
+        -------
+        Boolean
+        """
+
+        LOG.debug("Validating SAMRC config with given JSONSchema")
+        jsonschema.validate(config, schema)
+
+        LOG.debug("SAMRC looks valid!")
+        LOG.debug("Schema used: %s", schema)
+        LOG.debug("Config used: %s", config)
+
+        return True
```

Comment:
When is this false? Does `jsonschema.validate` through validation exceptions? 

The return here is used, consider removing.